fast-vm

```diff
@@ -966,9 +985,38 @@ case "$1" in
 				pmsg $P_ERROR "$3 not executable\n" $vm_number
 			fi
 		else
-			ssh -o GlobalKnownHostsFile=/dev/null -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "root@192.168.$SUBNET_NUMBER.$vm_number"
+			create_sshkey_if_needed
+			ssh -i $FASTVM_USER_CONF_DIR/id_fastvm -o GlobalKnownHostsFile=/dev/null -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "root@192.168.$SUBNET_NUMBER.$vm_number"
 		fi
 		;;
+	scp)
+		vm_number="$2"
+		if [ -z "$vm_number" ]; then pmsg $P_ERROR "VM number missing\n"; usage ssh; fi
+		if [[ $vm_number = *[\ ]* ]]; then handle_multiple_requests $@;	fi
+
+		check_vm_number "$vm_number" "inactive"
+		update_access_time "$vm_number"
+
+		wait_for_ssh "$vm_number"
+		sourcepath=$(echo "$3" | sed "s/\(.*\)\bvm:\(.*\)/\1root@192.168.$SUBNET_NUMBER.$vm_number:\2/")
+		destpath=$(echo "$4" | sed "s/\(.*\)\bvm:\(.*\)/\1root@192.168.$SUBNET_NUMBER.$vm_number:\2/")
+		create_sshkey_if_needed
+		scp -i $FASTVM_USER_CONF_DIR/id_fastvm -o GlobalKnownHostsFile=/dev/null -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no "$sourcepath" "$destpath"
+		;;
+	keydist)
+		vm_number="$2"
+		if [ -z "$vm_number" ]; then pmsg $P_ERROR "VM number missing\n"; usage ssh; fi
+		if [[ $vm_number = *[\ ]* ]]; then handle_multiple_requests $@;	fi
```

Comment:
I think that this function cannot be run in parallel so the `handle_multiple_request` should not be used here. I'm guessing that this is somehow connected with additional of SERIALIZE in the line 165. I would expect that if the operation requires output that cannot be automated (like here) then it wouldn't be possible to call it with multiple VMs.

Reply:
You are correct -- SERIALIZE permits you to type the password for each VM that's receiving the key.  This is no worse than having to invoke fast-vm repeatedly, and maybe a little bit nicer?  :)

git will update shortly... thank you again!